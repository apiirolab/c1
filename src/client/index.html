<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>