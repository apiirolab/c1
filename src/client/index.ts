export const info = 'Client Info';
