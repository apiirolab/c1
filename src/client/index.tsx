import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
