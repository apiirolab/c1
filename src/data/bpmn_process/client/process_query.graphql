query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
