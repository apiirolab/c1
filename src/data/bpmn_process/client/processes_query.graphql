query Processes($input: BpmnProcessesInput!) {
  processes(input: $input) {
    id
    name
  }
}
