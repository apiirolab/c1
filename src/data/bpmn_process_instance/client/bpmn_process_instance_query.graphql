fragment BpmnProcessResource on Resource {
  name
  type
  id
  content
}

fragment BpmnProcessComment on Comment {
  id
  date
  replyTo
  text
  user {
    name
  }
}

fragment BpmnProcessTask on BpmnTaskInstance {
  name
  dateStarted
  dateFinished
  performer {
    name
  }
  performerRoles
}

fragment BpmnProcessDefinition on BpmnProcess {
  name
  description
  type
  model
  resources {
    ...BpmnProcessResource
  }
}

query BpmnProcessInstanceQuery($id: String!) {
  bpmnProcessInstance(id: $id) {
    id
    status
    process {
      ...BpmnProcessDefinition
    }
    owner {
      name
    }
    tasks {
      ...BpmnProcessTask
    }
    comments {
      ...BpmnProcessComment
    }
    resources
    dateStarted
    dateFinished
  }
}
