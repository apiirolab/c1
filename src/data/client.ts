import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
