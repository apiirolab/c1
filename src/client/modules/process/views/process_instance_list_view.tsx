import * as React from 'react';

import { Header, List, ListProps, Message, SemanticICONS } from 'semantic-ui-react';
import styled, { StyledComponentClass } from 'styled-components';

import { QueryTypes } from 'data/client';
import { ProcessInstanceItem } from './process_instance_item_view';

const ProcessList: StyledComponentClass<ListProps, {}> = styled(List)`
  max-width: 800px !important;

  &&&&& .description {
    font-size: smaller;
  }
`;

const ResultInfo = styled.div`
  color: #999;
`;

type Props = {
  header: string;
  description: string;
  icon: SemanticICONS;
  processes: QueryTypes.BpmnProcessInstanceItem[];
  context: App.Context;
};

export const ProcessInstanceList: React.SFC<Props> = ({
  processes,
  context,
  icon,
  header,
  description
}) => (
  <React.Fragment>
    <ResultInfo>
      {processes.length === 1
        ? context.i18n`1 process available`
        : context.i18n`${processes.length.toString()} processes available`}
    </ResultInfo>
    <Header icon={icon} content={header} subheader={description} />
    <Choose>
      <When condition={processes && processes.length > 0}>
        <ProcessList divided relaxed>
          {processes.map(p => (
            <ProcessInstanceItem key={p.id} processInstance={p} context={context} />
          ))}
        </ProcessList>
      </When>
      <Otherwise>
        <Message>No results</Message>
      </Otherwise>
    </Choose>
  </React.Fragment>
);

ProcessInstanceList.displayName = 'ProcessInstanceList';
