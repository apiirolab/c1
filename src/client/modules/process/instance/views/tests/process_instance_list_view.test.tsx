import * as React from 'react';

import { create, render } from 'client/tests';

import { ProcessInstanceList } from '../process_instance_list_view';

import { createProcessInstanceList } from 'client/modules/process/common/tests/process_test_data';
import 'jest-styled-components';

describe('Process Instance', () => {
  describe('List', () => {
    const context = create.context();

    describe('Empty', () => {
      const component = (
        <div>
          <ProcessInstanceList
            header="Empty List"
            icon="tasks"
            description="This is the list of all your processes"
            processes={[]}
            context={context}
          />
        </div>
      );

      it('renders empty', () => {
        const root = render(component);
        expect(root).toMatchSnapshot();
      });

      return { component };
    });

    describe('Data', () => {
      const component = (
        <div>
          <ProcessInstanceList
            header="All Processes"
            icon="tasks"
            description="This is the list of all your processes"
            processes={createProcessInstanceList()}
            context={context}
          />
        </div>
      );

      it('renders correctly', () => {
        const list = render(component);
        expect(list).toMatchSnapshot();
      });

      return { component };
    });
  });
});
