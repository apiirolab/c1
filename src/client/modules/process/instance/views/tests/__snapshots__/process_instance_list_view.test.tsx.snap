// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Instance List Data renders correctly 1`] = `
<style>
.c2.c2.c2.c2.c2 {
  vertical-align: top !important;
  width: 25px;
}

.c3 {
  color: #999;
  margin: 6px 0px;
}

.c4 {
  margin: 3px 0px;
}

.c1 {
  max-width: 800px !important;
}

.c1.c1.c1.c1.c1 .description {
  font-size: smaller;
}

.c0 {
  color: #999;
}</style>

<div>
  <div
    className="c0"
  >
    5 processes available
  </div>
  <div
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="tasks icon"
    />
    <div
      className="content"
    >
      All Processes
      <div
        className="sub header"
      >
        This is the list of all your processes
      </div>
    </div>
  </div>
  <div
    className="ui divided relaxed list c1"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/process-1/view/information/1"
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="c3"
          >
            <div
              className="ui green tiny label"
              onClick={[Function]}
            >
              Running
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="c4"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Task Name" 2 months ago
          </div>
          <div
            className="c4"
          >
            <b>
              Waiting for: 
            </b>
             
            Purchasing to Complete Purchase since 2 months ago and IT Support or IT Delivery to Deliver Technology since 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/process-1/view/information/2"
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="c3"
          >
            <div
              className="ui green tiny label"
              onClick={[Function]}
            >
              Running
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="c4"
          >
            <b>
              Waiting for: 
            </b>
             
            Approver to Task Name since 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/process-1/view/information/3"
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="c3"
          >
            <div
              className="ui red tiny label"
              onClick={[Function]}
            >
              Aborted
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="c4"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Task Name" 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/process-1/view/information/4"
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="c3"
          >
            <div
              className="ui yellow tiny label"
              onClick={[Function]}
            >
              Paused
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="c4"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Task Name" 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/process-1/view/information/5"
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="c3"
          >
            <div
              className="ui grey tiny label"
              onClick={[Function]}
            >
              Finished
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
             · 
            Finished 2 months ago
          </div>
          <div
            className="c4"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Task Name" 2 months ago
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Process Instance List Empty renders empty 1`] = `
<style>
.c0 {
  color: #999;
}</style>

<div>
  <div
    className="c0"
  >
    0 processes available
  </div>
  <div
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="tasks icon"
    />
    <div
      className="content"
    >
      Empty List
      <div
        className="sub header"
      >
        This is the list of all your processes
      </div>
    </div>
  </div>
  <div
    className="ui message"
  >
    No results
  </div>
</div>
`;
